core/src/main/java/io/questdb/griffin/engine/groupby/GroupByLongHashSet.java

```diff
@@ -181,7 +182,9 @@ private long probe(long key, long index) {
             if (key == k) {
                 return -index - 1;
             }
-        } while (true);
+        } while (index != index0);
+
+        throw CairoException.critical(0).put("corrupt hash table");
```

Comment:
```suggestion
        throw CairoException.critical(0).put("corrupt long64 hash set state");
```

Reply:
Changed in 20ba60c6b4d95ff2aec27a0f99f028032b891afe